spec/std/http/server/handlers/static_file_handler_spec.cr

```diff
@@ -21,29 +21,94 @@ describe HTTP::StaticFileHandler do
     response.body.should eq(File.read("#{__DIR__}/static/test.txt"))
   end
 
-  context "with header If-Modified-Since" do
+  it "should add Etag header" do
+    response = handle HTTP::Request.new("GET", "/test.txt")
+    response.headers["Etag"].should match(/W\/"\d+"$/)
+  end
+
+  it "should add Last-Modified header" do
+    response = handle HTTP::Request.new("GET", "/test.txt")
+    response.headers["Last-Modified"].should eq(HTTP.rfc1123_date(File.info("#{__DIR__}/static/test.txt").modification_time))
+  end
+
+  context "with If-Modified-Since header" do
     it "should return 304 Not Modified if file mtime is equal" do
+      initial_response = handle HTTP::Request.new("GET", "/test.txt")
+
       headers = HTTP::Headers.new
-      headers["If-Modified-Since"] = HTTP.rfc1123_date(File.info("#{__DIR__}/static/test.txt").modification_time)
+      headers["If-Modified-Since"] = initial_response.headers["Last-Modified"]
+
       response = handle HTTP::Request.new("GET", "/test.txt", headers), ignore_body: true
       response.status_code.should eq(304)
-      response.headers["Last-Modified"].should eq(HTTP.rfc1123_date(File.info("#{__DIR__}/static/test.txt").modification_time))
+    end
+
+    it "should not return Content-Type if file mtime is equal" do
+      initial_response = handle HTTP::Request.new("GET", "/test.txt")
+
+      headers = HTTP::Headers.new
+      headers["If-Modified-Since"] = initial_response.headers["Last-Modified"]
+
+      response = handle HTTP::Request.new("GET", "/test.txt", headers), ignore_body: true
+      response.headers["Content-Type"]?.should eq(nil)
     end
 
     it "should return 304 Not Modified if file mtime is older" do
       headers = HTTP::Headers.new
       headers["If-Modified-Since"] = HTTP.rfc1123_date(File.info("#{__DIR__}/static/test.txt").modification_time + 1.hour)
       response = handle HTTP::Request.new("GET", "/test.txt", headers), ignore_body: true
+
       response.status_code.should eq(304)
-      response.headers["Last-Modified"].should eq(HTTP.rfc1123_date(File.info("#{__DIR__}/static/test.txt").modification_time))
     end
 
     it "should serve file if file mtime is younger" do
       headers = HTTP::Headers.new
       headers["If-Modified-Since"] = HTTP.rfc1123_date(File.info("#{__DIR__}/static/test.txt").modification_time - 1.hour)
-      response = handle HTTP::Request.new("GET", "/test.txt")
+      response = handle HTTP::Request.new("GET", "/test.txt", headers)
```

Comment:
Maybe add `ignore_body: false` to make it explicit that this is expecting content.